src/core/coap/coap.cpp

```diff
@@ -646,10 +644,7 @@ void Coap::ProcessReceivedRequest(Header &aHeader, Message &aMessage, const Ip6:
     Message *cachedResponse = NULL;
     otError error = OT_ERROR_NOT_FOUND;
 
-    if (mInterceptor != NULL)
-    {
-        SuccessOrExit(error = mInterceptor(aMessage, aMessageInfo, mContext));
-    }
+    SuccessOrExit(GetNetif().ApplyTmfFilter(aMessageInfo, GetNetif().GetMle()));
```

Comment:
I prefer keeping the `Interceptor` for it don't strengthen CoAP's dependence on `ThreadNetif`.